runtime/lua/vim/diagnostic.lua

```diff
@@ -637,6 +613,27 @@ end
 ---                * priority: (number, default 10) Base priority to use for signs. When
 ---                {severity_sort} is used, the priority of a sign is adjusted based on
 ---                its severity. Otherwise, all signs use the same priority.
+---                * sign_text: (table) Table of vim.diagnostic.severity symbol pairs.
+---                Text values in this table take priority over existing sign definitions
+---                but will not be prioritized over a sign_text field for an individual
+---                diagnostic. Individual diagnostic's sign_text can be set within a function
+---                passed to the filter parameter.
+---                * filter: (function) A function which will be run prior to determining the signs
+---                This function takes the diagnostics table as a parameter and can be used to filter
+---                or otherwise customize how signs will appear. Example
+---                <pre>
+---                  function(diagnostic)
+---                    -- Replace signs for diagnostics relating to <symbol> being unused with 'U'
+---                    if string.lower(diagnostic.message):find('unused') then
+---                       diagnostic.sign_text = 'U'
+---                    end
+---                    -- do not show signs for any INFO severity diagnostic
+---                    if diagnostic.severity == vim.diagnostic.severity.INFO then
+---                      return
+---                    end
+---                    return diagnostic
+---                  end
+---                </pre>
```

Comment:
Not *strictly* necessary, but I suggest running the `scripts/gen_vimdoc.py` script in the repo to generate the API docs with these updates. This lets us review the documentation changes as part of the PR.